### ISSUE TITLE
Losing some type data when sending messages to remote actors via the cluster

### ISSUE THREAD

**Issue by BrainSlugs83** [akkadotnet/akka.net#1534]
I have a class with a property of type `object` on it; it seems that if I set this property to `Guid.NewGuid()`, and send it to a remote actor system, it shows up as a `string` instead of as a `Guid`.

Even worse, if I set this property to the value of a custom object, the message just plain never arrives!

I'm guessing the serializer isn't outputting types for properties, and so the deserializer doesn't know anything about the serialized property except that it's of type object, and so it's guessing `string` as the type in the first case, and throws an exception in the second case.

**Comment by rogeralsing** [comment]
Json.NET have been giving us trouble for quite some time.
There is an alternative serializer in the contrib package here (not on Nuget yet).
https://github.com/akkadotnet/akka.net/tree/dev/src/contrib/serializers

Enable it using 

```
akka.actor {
    serializers {
        wire = " Akka.Serialization.WireSerializer, Akka.Serialization.Wire"
    }
    serialization-bindings {
      "System.Object" = wire
    }
}
```

That should do the trick

**Comment by BrainSlugs83** [comment]
Thanks Roger, I figured it was something like that.

In JSON.NET there are several options in the SerializationSettings for turning on type emitting that I think would fix bugs like this, and also ways to register custom serialization handlers for specific types and such.

Is there a way to latch in to the SerializationSettings that akka.net is using and configure them?

**Comment by BrainSlugs83** [comment]
For others running into this, I used a workaround:

I changed my messages with weak-typed "Object" properties into messages with strong-typed generic "T" properties.

I also declared an interface with all the same properties -- except for that strong typed one, I put it back as a weak typed object (I implement that property on the class explicitly [just returning the strong typed property] so as to not mess up the intellisense).

Then in the handler, I just "Receive<IMyInterface>()".

This works because the property is strong typed via the generic class's definition when deserializing it.

Basically, before:

    public class MyMessage
    {
        public object Message { get; private set; }

        // ...
    }

After:

    public interface IMyMessage
    {
        T Message { get; }

        // ...
    }

    public class MyMessage<T>: IMyMessage
    {
        public T Message { get; private set; }
        object IMyMessage.Message { get { return this.Message; } }

        // ...
    }

And that works. -- The strong type helps it get serialized, and the interface allows you to still handle all generic versions of the class in a typed receiver...

It's still a super annoying bug that I think we could just configure our way out of (i.e. it should be pretty easy to configure JSON.NET to serialize the type information) -- but this is just a work around that I'm sharing in case it helps anyone.
